### PR TITLE
Frontier console node admin interface

### PR DESCRIPTION
**Pull request by zelig**
based on https://github.com/ethereum/go-ethereum/pull/484
This PR implements a node admin interface as speced here
https://github.com/ethereum/go-ethereum/wiki/Frontier-Console
accessible via the `admin` object in the Frontier JS Console and VM

p2p server and discover/table exposes Node to support nodeInfo()
eth implements NodeInfo() and PeersInfo() to support the node admin interface

 see https://github.com/ethereum/go-ethereum/issues/459

### CODE REVIEW

**Comment by obscuren** [comment]
Fix the tests please

File: eth/backend.go [hunk 1]
```diff
@@ -219,6 +219,64 @@ func New(config *Config) (*Ethereum, error) {
...
+	}
+}
+
+type PeerInfo struct {
+	ID            string
+	Name          string
+	Caps          string
+	RemoteAddress string
+	LocalAddress  string
+}
```
Comment (fjl) [inline eth/backend.go]: Why is this type necessary? Can't it just return peers directly?

File: eth/backend.go [hunk 2]
```diff
@@ -251,6 +309,7 @@ func (s *Ethereum) StateDb() common.Database            { return s.stateDb }
 func (s *Ethereum) ExtraDb() common.Database            { return s.extraDb }
 func (s *Ethereum) IsListening() bool                    { return true } // Always listening
 func (s *Ethereum) PeerCount() int                       { return s.net.PeerCount() }
+func (s *Ethereum) PeerInfo() int                        { return s.net.PeerCount() }
```
Comment (fjl) [inline eth/backend.go]: What is this about? Is that intended?

**Comment by obscuren** [comment]
:+1:

File: eth/backend.go (hunk 2 above)
Comment (zelig) [inline eth/backend.go]: Absolutely not, thanks for spotting. Will fix in https://github.com/ethereum/go-ethereum/issues/512

**Pull request merged**